tensorflow_probability/python/positive_semidefinite_kernels/polynomial.py

```diff
@@ -18,6 +18,10 @@
 from __future__ import division
 from __future__ import print_function
 
+
+from functools import reduce
```

Comment:
as a style rule we import only modules (or submodules), but not individual exported items from modules. please import functools here and use functools.reduce below